test-coverage-enforcer/src/main/java/org/wso2/testcoverageenforcer/Maven/POM/MavenPom.java

```diff
@@ -0,0 +1,188 @@
+
+        if (!(this.pomFile.getModules().size() > 0)) { //Check with profiles for children
+            for (Profile eachProfile : this.pomFile.getProfiles()) {
+                if (eachProfile.getModules().size() > 0) {
+                    return true;
+                }
+            }
+        } else {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Get a list of child modules to examine each
+     *
+     * @return A list of child modules if available. Empty list if not.
+     * @throws PomFileReadException Error while reading the pom
+     */
+    public List<ChildPom> getChildren() throws PomFileReadException {
+
+        List<ChildPom> childPomList = new ArrayList<>(0);
+        /*
+        Sometimes modules are available under <profiles> tag. Not under <modules> tag.
+        In that case, look under <profiles> tag for modules and add coverage check for
+        all available modules
+         */
+        List<String> modules = this.pomFile.getModules();
+        if (modules.size() == 0) {
+            List<Profile> profilesList = this.pomFile.getProfiles();
+
+            //Both <modules> and <profiles> are not present, pom does not have children
+            if (profilesList.size() == 0) return null;
+            List<List<String>> modulesList = new ArrayList<>();
+            for (Profile eachProfile : profilesList) {
+                List<String> modulesInProfile = eachProfile.getModules();
+                if (modulesInProfile.size() > 0) {
+                    modulesList.add(modulesInProfile);
+                }
+            }
+
+            //No modules are present in available profiles
+            if (modulesList.size() == 0) return null;
+            modules = modulesList.stream().flatMap(x -> x.stream()).collect(Collectors.toList());
+        }
+        for (String eachChildPomPath : modules) {
+            if (eachChildPomPath.contains(Constants.Child.CHILD_NAME_TESTS_INTEGRATION)) {
+                log.info("Skipping " + this.pomFilePath.replace(Constants.POM_NAME, "") + eachChildPomPath + ". Integration test module");
+                continue;    //Neglect the module 'tests-integration'
+            } else if (eachChildPomPath.contains(Constants.Child.CHILD_NAME_OSGI)) {
+                log.info("Skipping " + this.pomFilePath.replace(Constants.POM_NAME, "") + eachChildPomPath + ". OSGI test module");
+                continue;
+            }
+            childPomList.add(new ChildPom(this.pomFilePath.replace(Constants.POM_NAME,
+                    "") + eachChildPomPath));
+        }
+        return childPomList;
+    }
+
+    /**
+     * Check whether this pom directory should contain tests by searching for src/test folder
+     *
+     * @return True if src/test exists. Otherwise False.
+     */
+    public Boolean hasTests() {
+
+        File testFolder = new File(this.pomFilePath.replace(Constants.POM_NAME,
+                "") + Constants.TEST_FOLDER);
```

Comment:
Formatting issue

Reply:
Fixed in 2bc0618a39bb056608ff116edf53bc2b3f4e8099.